web/styles/left_sidebar.css

```diff
@@ -796,6 +839,42 @@ li.top_left_scheduled_messages {
     grid-template-areas:
         "starting-offset starting-anchor-element row-content markers-and-controls ending-anchor-element ending-offset"
         ".               .                       filter-box  filter-box           filter-box            .            ";
+
+    .masked_unread_count {
+        grid-area: markers-and-controls;
+    }
+
+    &.hide_unread_counts {
+        .unread_count,
+        .unread_count.hide,
+        .masked_unread_count {
+            display: none;
+        }
+
+        /* When an empty unread count is hidden,
+                    hide the masked unread count, too. */
+        .unread_count.hide + .masked_unread_count {
+            display: none;
+        }
+
+        .masked_unread_count {
+            display: block;
+            /* Hold space enough so single-digit
+                        unreads don't shift on hover. */
+            margin: 0 3px;
+        }
+
+        &:hover {
+            .unread_count {
+                display: block;
+            }
+
+            .hide,
+            .masked_unread_count {
+                display: none;
+            }
+        }
+    }
 }
```

Comment:
This CSS refactoring is quite hard to understand; can you use some preparatory commits with proper commit messages to explain it?